src/Evolution/DiscontinuousGalerkin/Actions/InternalMortarDataImpl.hpp

```diff
@@ -230,53 +229,110 @@ void internal_mortar_data_impl(
+    // dg_package_data. However, if we need to project, then we hae to use the
+    // mortar_projection_buffer that was passed in.
+    if (neighbors_in_direction.size() == 1) {
+      const auto& neighbor = *neighbors_in_direction.begin();
+      const auto mortar_id = std::pair{direction, neighbor};
+      const auto& mortar_mesh = mortar_meshes.at(mortar_id);
+      const auto& mortar_size = mortar_sizes.at(mortar_id);
 
-    const double max_abs_char_speed_on_face = detail::dg_package_data<System>(
+      // Have to use mortar_projection_buffer
+      if (Spectral::needs_projection(face_mesh, mortar_mesh, mortar_size)) {
+        // The face mesh will be assigned below along with ensuring the size of
+        // the mortar data is correct
+        packaged_data.set_data_ref(mortar_projection_buffer->data(),
+                                   total_face_size);
+      } else {
+        // Can use the local_mortar_data
+        auto& local_mortar_data_opt =
+            mortar_data_ptr->at(mortar_id).local_mortar_data();
+        // If this isn't the first time, set the face mesh
+        if (LIKELY(local_mortar_data_opt.has_value())) {
+          local_mortar_data_opt->first = face_mesh;
+        } else {
+          // Otherwise we need to initialize the pair. If we don't do this, then
+          // the DataVector will be non-owning which we don't want
+          local_mortar_data_opt =
+              std::optional{std::pair{face_mesh, DataVector{}}};
+        }
+
+        // Always set the time step ID
+        mortar_data_ptr->at(mortar_id).time_step_id() = temporal_id;
+
+        auto& local_mortar_data = local_mortar_data_opt->second;
+
+        // Do a destructive resize to account for potential p-refinement
+        local_mortar_data.destructive_resize(total_face_size);
+
+        ASSERT(local_mortar_data.size() == total_face_size,
+               "AhHHhhhhHHHHhHhhhHHh");
+        packaged_data.set_data_ref(local_mortar_data.data(),
+                                   local_mortar_data.size());
+      }
+    } else {
+      // In this case, we have multiple neighbors in this direction so all will
+      // need to project their data which means we use the
+      // mortar_projection_buffer to calculate the dg_package_data
+      packaged_data.set_data_ref(mortar_projection_buffer->data(),
+                                 total_face_size);
+    }
+
+    detail::dg_package_data<System>(
         make_not_null(&packaged_data), boundary_correction, fields_on_face,
         get<evolution::dg::Tags::NormalCovector<Dim>>(
             *normal_covector_and_magnitude_ptr->at(direction)),
         face_mesh_velocity, dg_package_data_projected_tags{},
         package_data_volume_args...);
-    (void)max_abs_char_speed_on_face;
 
     // Perform step 3
+    // This will only do something if
+    //  a) we have multiple neighbors in this direction
+    //  b) the one (and only) neighbor in this direction needed projection
```

Comment:
Add an explicit "or" connecting these cases, so it is clear it isn't supposed to be "and".